### DISCUSSION TITLE
Problem with two almost identical command

### DISCUSSION THREAD

**Discussion by asmaaharfoush** [neperfepx/neper#226] [Q&A] [answered]
Hello,

I wrote a small code that changes the morpho, rotation, and ratio. It does working pretty well except for some values such this command that does not generate a figure:
```
neper -T -n from_morpho -morpho 'diameq:normal(0.07,0.01)' -o 4 -id 4 -dim 2 -domain 'square(1,1):rotate(0,0,1,45):scale(0.5773502691896258,1.7320508075688772,1)' -transform 'scale(1.7320508075688772,0.5773502691896258,1),rotate(0,0,1,-45 )'
neper -V 4.tess -imagesize 256:256 -dataedgerad 0.001 -print 4
```
I don't have any explanation why the previous command does not work however the former and latter commands working properly:
```
neper -T -n from_morpho -morpho 'diameq:normal(0.07,0.01)' -o 3 -id 3 -dim 2 -domain 'square(1,1):rotate(0,0,1,25):scale(0.5773502691896258,1.7320508075688772,1)' -transform 'scale(1.7320508075688772,0.5773502691896258,1),rotate(0,0,1,-25 )'
neper -V 3.tess -imagesize 256:256 -dataedgerad 0.001 -print 3
```

```
neper -T -n from_morpho -morpho 'diameq:normal(0.07,0.01)' -o 5 -id 5 -dim 2 -domain 'square(1,1):rotate(0,0,1,65):scale(0.5773502691896258,1.7320508075688772,1)' -transform 'scale(1.7320508075688772,0.5773502691896258,1),rotate(0,0,1,-65 )'
neper -V 5.tess -imagesize 256:256 -dataedgerad 0.001 -print 5
```

Do you have any explanation for that, thank you!

**Comment by rquey** [comment, marked as answer]
This was a bug, now fixed (version 4.1.3-12). Because the domain is rotated by 45° (and scaled), Neper had troubles finding which face is which from the face equations, to retrieve the face labels. This is now properly handled, and, when this configuration occurs, the labels are set to `f1`, `f2`, etc. (some of them were previously empty, which is what was causing issues when parsing the tess file).

To retrieve the usual `x0`, `x1`, etc. for a cubic domain, use `-transform resetlabels`.  This is a patch to your specific configuration (rotated / scaled domain) and does not need to be used in general.
```
neper -T -n from_morpho -morpho 'diameq:normal(0.07,0.01)' -o 4 -id 4 -dim 2 -domain 'square(1,1):rotate(0,0,1,45):scale(0.5773502691896258,1.7320508075688772,1)' -transform 'scale(1.7320508075688772,0.5773502691896258,1),rotate(0,0,1,-45),resetlabels'
neper -V 4.tess -imagesize 256:256 -dataedgerad 0.001 -print 4
```

**Reply by asmaaharfoush** [reply]
Thanks for your reply, I did not find version 4.1.3-12 in either https://neper.info/generation.html or https://github.com/rquey/neper/releases?fbclid=IwAR1GO9qWU1z0FEKeRUHSRdRhd0JjiPy3gI9s1kCnd1Ob3rGt2q4mbIkx6t4.

When I added -transform resetlabels using my current version 4.1.2 doesn't work, do you have any alternative solution for this that can be work with this version?

Do you have any suggestions where I can find the latest version of neper?

Thank you!

**Reply by rquey** [reply]
`4.1.3-12` is an intermediary version, not a _release_. You can access from https://github.com/rquey/neper, preferably by cloning the repository, or as a zip file (see _Code_).

**Reply by asmaaharfoush** [reply]
Thank you for your reply, I tried to uninstall Neper to install the new version that you recommended using this command "Sudo apt-get remove neper-4.1.2", however, it does not work and I did find any documentation or instruction on how to uninstall Neper. So do you have any recommendations on how I can install this version and uninstall the old version?
Thank you!

**Reply by rquey** [reply]
When you install the new version, it will just override the previous version.

Otherwise, to uninstall, simply find where your Neper binary is (e.g. using `which neper`) and remove it.

**Reply by asmaaharfoush** [reply]
Thank you, it works!
